In [28]:
import pandas as pd

file_path = '/Users/ludovica/Documents/Bachelor/BA5/Modeling Lab/4-Molecules.csv'
df = pd.read_csv(file_path)

df.columns = df.columns.str.strip()

# Convert relevant columns to numeric
df['Mannhold LogP'] = pd.to_numeric(df['Mannhold LogP'], errors='coerce')
df['Molecular Weight'] = pd.to_numeric(df['Molecular Weight'], errors='coerce')
df['Lipinski\'s Rule of Five'] = pd.to_numeric(df['Lipinski\'s Rule of Five'], errors='coerce')

# Set 1: Rows where Lipinski's Rule of Five != 0 and Mannhold LogP <= 5
subset1 = df[(df['Lipinski\'s Rule of Five'] != 0) & (df['Mannhold LogP'] <= 5)]

# Set 2: Rows where Lipinski's Rule of Five == 0 and Molecular Weight <= 500
subset2 = df[(df['Lipinski\'s Rule of Five'] == 0) & (df['Molecular Weight'] <= 500)]

# 10 first  results for Subset 1
print("\nFirst 10 Rows where Lipinski's Rule of Five != 0 and Mannhold LogP <= 5:")
for index, row in subset1.head(10).iterrows():
    print("Row {}: Lipinski's Rule of Five = {}, Mannhold LogP = {}".format(
        index, row['Lipinski\'s Rule of Five'], row['Mannhold LogP']
    ))

# 10 first  results for Subset 2
print("\nFirst 10 Rows where Lipinski's Rule of Five == 0 and Molecular Weight <= 500:")
for index, row in subset2.head(10).iterrows():
    print("Row {}: Lipinski's Rule of Five = {}, Molecular Weight = {}".format(
        index, row['Lipinski\'s Rule of Five'], row['Molecular Weight']
    ))

subset1.to_csv('lipinski_compliant_molecules.csv', index=False)
subset2.to_csv('lipinski_violation_molecules.csv', index=False)



Columns in the DataFrame:
Index(['row ID', 'Molecule', 'Molecule name', 'Mannhold LogP',
       'Atomic Polarizabilities', 'Aromatic Atoms Count',
       'Aromatic Bonds Count', 'Element Count', 'Bond Polarizabilities',
       'Bond Count', 'Eccentric Connectivity Index', 'Fragment Complexity',
       'VABC Volume Descriptor', 'Hydrogen Bond Acceptors',
       'Hydrogen Bond Donors', 'Largest Chain', 'Largest Pi Chain',
       'Petitjean Number', 'Rotatable Bonds Count', 'Lipinski's Rule of Five',
       'Topological Polar Surface Area',
       'Vertex adjacency information magnitude', 'Molecular Weight', 'XLogP',
       'Zagreb Index', 'Molecular Formula', 'Formal Charge',
       'Formal Charge (pos)', 'Formal Charge (neg)', 'Heavy Atoms Count',
       'Molar Mass', 'SP3 Character', 'Rotatable Bonds Count (non terminal)'],
      dtype='object')

First 10 Rows where Lipinski's Rule of Five != 0 and Mannhold LogP <= 5:
Row 3: Lipinski's Rule of Five = 3, Mannhold LogP = 0.25
Row 11: Lip

Lipinski's rule of five helps predict if a bioactive molecule such as a drug is likely to be orally active in humans, it helps identifying hydrophobic compounds that can more easily penetrate membranes. A value different from 0 indicates one of the following is being broken and the compound is not orally active. 
    1. Molecular Weight <= 500 Da
    2. logP<=5
    3. hydrogen bond donors<=5
    4. hydrogen bond acceptors <=10.
The first subset identifies the compounds that are not orally active i.e. the Lipinski's value is different from 0, but whose problem is not the partition coefficient i.e. logP<=5.
The second subset identifies the compounds that are predicted to be orally active i.e. Lipiski's value is 0 (no rules are broken) and the molecular weight is lower than 500 Da. This subset helped understand what the value 0 meant: wether it indicated 0 drug activity or 0 rules broken. 

In [29]:
import pandas as pd

file_path = '/Users/ludovica/Documents/Bachelor/BA5/Modeling Lab/4-Molecules.csv'
df = pd.read_csv(file_path)

df.columns = df.columns.str.strip()

# Convert the SP3 Character column to numeric
df['SP3 Character'] = pd.to_numeric(df['SP3 Character'], errors='coerce')

# Calculate statistics
max_sp3 = df['SP3 Character'].max()
min_sp3 = df['SP3 Character'].min()
mean_sp3 = df['SP3 Character'].mean()
std_sp3 = df['SP3 Character'].std()

print("SP3 Character Statistics:")
print(f"Maximum: {max_sp3}")
print(f"Minimum: {min_sp3}")
print(f"Mean: {mean_sp3}")
print(f"Standard Deviation: {std_sp3}")


SP3 Character Statistics:
Maximum: 1.0
Minimum: 0.0
Mean: 0.14771763841236166
Standard Deviation: 0.10050263593470635


Among the 15000 circa molecules the sp3 character statistics reveal the max reached is 1, meaning the molecule is likely to have sp3 hybridized orbitals, and 0 it is not likely. The mean is 0.15 so most of the molecules are less likely to be hybridized. Finally the standard deviation indicates that the 